### PR TITLE
package_index: FortranCallGraph and Serialbox

### PR DESCRIPTION
**Pull request by dev-zero**
While looking for a callgraph generator I stumbled over the FortranCallGraph project (a static callgrapher), from there you can find the FortranTestGenerator (unittest framework based on Capture & Replay via Serialbox), and the serialization library Serialbox (from Swiss Meteo).
The FortranTestGenerator I'm not yet adding due to lack of documentation example code.

### CODE REVIEW

File: _data/package_index.yml [hunk 1]
```diff
@@ -188,6 +188,13 @@
   tags: unit testing
   license: BSD 2-clause
 
+- name: FortranCallGraph
+  url: https://github.com/fortesg/fortrancallgraph
```
Comment (LKedward) [inline _data/package_index.yml:192]:
```suggestion
  github: fortesg/fortrancallgraph
```

File: _data/package_index.yml [hunk 2]
```diff
@@ -386,6 +393,12 @@
   tags: 
   license: MIT
 
+- name: Serialbox
+  url: https://github.com/GridTools/serialbox
```
Comment (LKedward) [inline _data/package_index.yml:397]:
```suggestion
  github: GridTools/serialbox
```

File: _data/package_index.yml [hunk 3]
```diff
@@ -386,6 +393,12 @@
...
+  description: A serialization library and tools for C/C++, Python3 and Fortran
+  categories: io
+  tags:
+  license: BSD
```
Comment (LKedward) [inline _data/package_index.yml:401]:
```suggestion
  license: "BSD 2 Clause"
```

File: _data/package_index.yml [hunk 4]
```diff
@@ -386,6 +393,12 @@
   tags: 
   license: MIT
 
+- name: Serialbox
+  url: https://github.com/GridTools/serialbox
+  description: A serialization library and tools for C/C++, Python3 and Fortran
+  categories: io
+  tags:
```
Comment (LKedward) [inline _data/package_index.yml:400]:
```suggestion
  tags: validation
```

**Comment by dev-zero** [comment]
@LKedward thanks, done

**Comment by LKedward** [comment]
#build_preview

**Comment by LKedward** [comment]
Packages and their badges are rendering properly 👍 .
I'm happy that these packages meet the criteria for listing so I'll merge.
#delete_preview

**Comment by milancurcic** [comment]
Just so I understand for future packages, this one satisfies the relevance criterion:

> Relevance: the package must be primarily implemented in Fortran or provide a complete Fortran interface to an existing package or be purposed solely towards software development in Fortran.

only because it's "purposed solely towards software development in Fortran", correct?

I ask because when I first took a looked at the repo, I saw that it's a Python package with only target code samples in Fortran, so it gave me pause. 

So, the 3rd test in the relevance criterion basically allows any non-Fortran tool that is somehow to be used for Fortran development. Another obvious example is FORD. Am I interpreting this correctly?

**Comment by LKedward** [comment]
> So, the 3rd test in the relevance criterion basically allows any non-Fortran tool that is somehow to be used for Fortran development. Another obvious example is FORD. Am I interpreting this correctly?

Yes, this is my reasoning. The third criterion was written this way precisely to allow FORD and similar utilities.
Ultimately fortran-lang should probably list development tools and utilities like FORD separately somewhere, but for now they allowed as 'packages' just as a convenience.

**Comment by certik** [comment]
I am a little worried that we are opening a pandora box with this ---- it seems we should also list any Fortran related tools, such as LFortran, GFortran, Flang, all kinds of plugins for VSCode etc. 

Maybe we can have a dedicated tag for these.

**Comment by LKedward** [comment]
> I am a little worried that we are opening a pandora box with this...

I agree, which is why a separate site page may be more suitable. Perhaps the _Compiler_ page becomes a more general page with compilers and development tools?

**Comment by certik** [comment]
I think we should list all good related tools for Fortran somehow. Including things like debuggers and optimization tools if they work with Fortran. Also currently we only list open source tools (but for compilers we want to list the commercial ones too).

So it's just a question where. For now maybe we can have a tag or category for "tools" and list all these open source tools there.

**Pull request merged**